IoTuring/Configurator/ConfiguratorIO.py

```diff
@@ -0,0 +1,144 @@
+    def linuxFolderPath(self):
+        return os.environ["XDG_CONFIG_HOME"] if "XDG_CONFIG_HOME" in os.environ else os.path.join(os.environ["HOME"], ".config")
+    
+    def envvarFolderPath(self):
+        return os.environ[CONFIG_PATH_ENV_VAR] if CONFIG_PATH_ENV_VAR in os.environ else None
+    
+    # In versions prior to 2022.12.2, the configurations file was stored in the same folder as this file
+    def oldFolderPath(self):
+        return os.path.dirname(inspect.getfile(ConfiguratorIO))
+    
+    def checkConfigurationFileInOldLocation(self):
+        """ Should be called at the beginning of the program when in -c mode.
+            Checks if a config. file is in the old folder path; if so ask if user wants to copy it
+            into new location, instead of rewriting it from zero.
+            If yes, copy it; otherwise leave it there and write a don't move file to remmeber the 
+            choose in the next times. 
+            This check is not done if old path = current chosen path 
+        """
+        
+        # This check is not done if old path = current chosen path (also check if ending slash is present)
+        if self.oldFolderPath() == self.getFolderPath() or self.oldFolderPath() == self.getFolderPath()[:-1]:
+            return
+        
+        # Exit check if no config. in old directory or if there is also the dont move file with it.
+        configuration_file_exists = os.path.exists(os.path.join(self.oldFolderPath(), CONFIGURATION_FILE_NAME))
+        dont_move_file_exists = os.path.exists(os.path.join(self.oldFolderPath(), DONT_MOVE_FILE_FILENAME))
+        if not configuration_file_exists or (configuration_file_exists and dont_move_file_exists):
+            return
+        
+        response = input("A configuration file has been found in the old location. Do you want to move it to the new location? (y/n): ")
+        response = True if response=="y" or response=="Y" else False
+        # Then ask to move it
+        if response:
+            # copy file from old to new location
+            os.rename(os.path.join(self.oldFolderPath(), CONFIGURATION_FILE_NAME), os.path.join(self.getFolderPath(), CONFIGURATION_FILE_NAME))
+            self.Log(self.LOG_MESSAGE, "Copied into \"" + os.path.join(self.getFolderPath(), CONFIGURATION_FILE_NAME) + "\"")
+        else:
+            # create dont move file
+            with open(os.path.join(self.oldFolderPath(), DONT_MOVE_FILE_FILENAME), "w") as f:
+                f.write("This file is here to remember you that you don't want to move the configuration file into the new location. If you want to move it, delete this file and \
+                    run the script in -c mode.")
+            self.Log(self.LOG_MESSAGE, "You won't be asked again. If you want to move the configuration file, delete \"" + os.path.join(self.oldFolderPath(), DONT_MOVE_FILE_FILENAME) + "\" and run the script in -c mode.")
```

Comment:
A note should be added that a new empty configuration will be created, not the old location will be remembered.